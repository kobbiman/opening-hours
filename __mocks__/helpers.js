/* eslint-disable react/prop-types */
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { theme } from '@lib/theming'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { render } from '@testing-library/react'

export const testRenderer = el => renderer.create(el).toJSON()

export const ThemedComponent = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const renderWithTheme = component => <ThemedComponent>{component}</ThemedComponent>

export const toHaveStyleRule = (component, property, value, options) => {
  expect(testRenderer(component)).toHaveStyleRule(property, value, options)
  expect(mount(component)).toHaveStyleRule(property, value, options)
  expect(render(component).container.firstChild).toHaveStyleRule(property, value, options)
}

export const shouldRenderProp = (component, as = 'div') => {
  const Component = component
  const id = 'testId'
  const { container } = render(<Component id={id} as={as} />)
  expect(container.querySelector(as).id).toEqual(id)
}

export const shouldRenderSomeText = (component, as = 'div') => {
  const Component = component
  const children = 'Text'
  const { container, queryByText } = render(<Component as={as}>{children}</Component>)
  const { childNodes } = container.querySelector(as)
  expect(childNodes).toHaveLength(1)
  expect(queryByText(children)).not.toBeNull()
}

export const mockRawOpeningHours = {
  monday: [],
  tuesday: [],
  wednesday: [],
  thursday: [],
  friday: [
    {
      type: 'open',
      value: 64800,
    },
  ],
  saturday: [
    {
      type: 'close',
      value: 3600,
    },
    {
      type: 'open',
      value: 32400,
    },
    {
      type: 'close',
      value: 39600,
    },
    {
      type: 'open',
      value: 57600,
    },
    {
      type: 'close',
      value: 86399,
    },
  ],
}

export const mockFormattedOpeningHours = [
  { key: 'monday', value: [] },
  { key: 'tuesday', value: [] },
  { key: 'wednesday', value: [] },
  { key: 'thursday', value: [] },
  { key: 'friday', value: [['6 PM', '1 AM']] },
  {
    key: 'saturday',
    value: [
      ['9 AM', '11 AM'],
      ['4 PM', '11:59 PM'],
    ],
  },
]
