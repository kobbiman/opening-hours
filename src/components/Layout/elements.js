import styled, { css } from 'styled-components'
import { Box } from 'rebass/styled-components'

const baseStyle = css`
  flex-shrink: 0;
`

export const Wrapper = styled.article`
  min-height: 100%;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  background-color: ${props => props.theme.colors.background};

  /* header, */
  footer {
    ${baseStyle};
  }
`
export const PageHeader = styled(Box)`
  ${baseStyle};
  position: sticky;
  top: 0;
  z-index: 1250;
`

export const PageBody = styled(Box)`
  ${baseStyle};
  flex-grow: 1;
`
