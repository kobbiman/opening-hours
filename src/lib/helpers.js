import PropTypes from 'prop-types'

const SECONDS_IN_HOUR = 3600

export const formatTime = value => {
  const hours = Math.floor(value / SECONDS_IN_HOUR)
  const minutes = Math.floor((value % SECONDS_IN_HOUR) / 60)

  const time = [hours % 12 || 12]

  if (minutes > 0) {
    time.push(`0${minutes}`.slice(-2))
  }

  const amOrPm = hours < 12 ? 'AM' : 'PM'
  return [time.join(':'), amOrPm].join(' ')
}

/**
 *
 * @param {Object} dayValues
 * @param {Object} nextDayValues
 */

const parseDay = (dayValues, nextDayValues) => {
  if (dayValues.length === 0) return dayValues

  const entries = []

  for (let index = 0; index < dayValues.length; index++) {
    const { value, type } = dayValues[index]
    const entryTime = formatTime(value)

    if (type === 'open') {
      // Add a new sub array if it's not the last entry
      if (index < dayValues.length - 1) {
        entries.push([entryTime])
      } else {
        // If it's 'open' and the last entry , find first 'close' in the next day
        entries.push([entryTime, formatTime(nextDayValues[0].value)])
      }
    } else {
      // If the first value of the array is 'close', do nothing as its irrelevant for the day context, otherwise
      // add entry value to last sub array in array
      if (index !== 0) {
        entries[entries.length - 1].push(entryTime)
      }
    }
  }

  return entries
}

export const formatOpeningHours = data => {
  if (!data) return []
  const keys = Object.keys(data)
  const formattedDays = keys.map((key, i) => {
    const dayData = data[key]

    // Get next day value, if index out of bounds return first item
    const nextDay = data[keys[i + 1]] || data[keys[0]]
    return { key, value: parseDay(dayData, nextDay) }
  })

  return formattedDays
}

export const weekdayName = new Date().toLocaleDateString('en', { weekday: 'long' }).toLowerCase()

export const TuppleChecker = (...validators) =>
  PropTypes.arrayOf((_, index) => {
    const currentValidators = validators.filter((v, i) => i === index)
    if (currentValidators.length <= 0) return true
    const [currentValidator] = currentValidators

    return currentValidator
  })
