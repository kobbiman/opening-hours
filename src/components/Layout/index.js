/**
 * Layout
 */

import React from 'react'
import PropTypes from 'prop-types'
import DefaultHeader from '@components/Header'
import { Wrapper, PageHeader, PageBody } from './elements'

const Noop = () => <></>

export function Layout({ children, bg, header: Header = DefaultHeader, footer: PageFooter = Noop }) {
  return (
    <Wrapper>
      <PageHeader>
        <Header />
      </PageHeader>
      <PageBody bg={bg} as="main">
        {children}
      </PageBody>
      <PageFooter />
    </Wrapper>
  )
}

Layout.propTypes = {
  header: PropTypes.func,
  footer: PropTypes.func,
  children: PropTypes.node.isRequired,
  bg: PropTypes.string,
}

export default Layout
