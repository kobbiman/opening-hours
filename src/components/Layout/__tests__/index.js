import React from 'react'
import renderer from 'react-test-renderer'
import { ThemedComponent } from '__mocks__/helpers'

import Layout from '../index.js'

describe('<Layout />', () => {
  it('should match the snapshot', () => {
    const component = (
      <ThemedComponent>
        <Layout>test</Layout>
      </ThemedComponent>
    )
    const renderedComponent = renderer.create(component).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})
