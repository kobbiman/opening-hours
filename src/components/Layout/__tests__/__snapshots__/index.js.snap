// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match the snapshot 1`] = `
.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 8px;
  padding-bottom: 8px;
  font-weight: 700;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #202125;
  background-color: transparent;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding-left: 12px;
  padding-right: 12px;
  width: 100%;
}

.c0 {
  min-height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #F8F8F8;
}

.c0 footer {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1250;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media (min-width:576px) {
  .c2 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c2 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c2 {
    max-width: 960px;
  }
}

@media (min-width:1280px) {
  .c2 {
    max-width: 1264px;
  }
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width="100%"
    >
      <div
        className="c3"
        color="text"
      >
        <div
          className="c4"
          fontWeight="bold"
        >
          WoltApp
        </div>
      </div>
    </div>
  </div>
  <main
    className="c5"
  >
    test
  </main>
</article>
`;
